src/inference/include/openvino/runtime/properties.hpp

```diff
@@ -142,7 +142,19 @@ class Property : public util::BaseProperty<T, mutability_> {
 
         template <typename U,
                   typename std::enable_if<!std::is_same<typename std::decay<U>::type, std::string>::value &&
-                                              !std::is_convertible<V, std::string>::value,
+                                              !std::is_convertible<V, std::string>::value && std::is_signed<V>::value,
+                                          bool>::type = true>
+        explicit operator U() {
+            if (std::is_unsigned<U>::value) {
+                if (value < 0)
+                    throw ov::Exception{"Expecting positive value for unsigned type"};
+            }
+            return value;
+        }
```

Comment:
can we also handle case of `constexpr` ?
E.g. `ov::hint::infer_request(-1)` will give compile time error

Reply:
from my point , yes compile runtime error is preferred, otherwise, each plugin has to make justification on the possible invalid value converted implicitly from a negative int value.
But currently there are many cases using like ov::hint::infer_request(-1) as invalid property inputs, probably we shall remove these?

Reply:
agree to remove 
BTW, `num_requests` is `uint32_t`, so when you specify `-1`, do not you have compile warning at least?